### ISSUE TITLE
Selecting an array literal of objects of projections merges the objects

### ISSUE THREAD

**Issue by garyb** [quasar-analytics/quasar#2064]
Using this data on a MongoDB mount:
``` json
{ "location": "whatever", "coords": [1, 1] }
{ "location": "somewhere", "coords": [20, 30] }
{ "location": "nowhere", "coords": [0, 0] }
```

With this query:
``` sql
SELECT [{ "lat": t.coords[0] }, { "lon": t.coords[1] }] AS arr FROM `/test/locations.json` AS t
```

Results in:
``` json
[
{ "arr": [ { "lat": 1.0, "lon": 1.0 }, { "lat": 1.0, "lon": 1.0 } ] },
{ "arr": [ { "lat": 20.0, "lon": 30.0 }, { "lat": 20.0, "lon": 30.0 } ] },
{ "arr": [ { "lat": 0.0, "lon": 0.0 }, { "lat": 0.0, "lon": 0.0 } ] }
]
```

If we use a constant for one of the key values, the behaviour is correct:
``` sql
SELECT [{ "lat": t.coords[0] }, { "lon": 0 }] AS arr FROM `/test/locations.json` AS t
```
``` json
[
{ "arr": [ { "lat": 1.0 }, { "lon": 0.0 } ] },
{ "arr": [ { "lat": 20.0 }, { "lon": 0.0 } ] },
{ "arr": [ { "lat": 0.0 }, { "lon": 0.0 } ] }
]
```

**Comment by sellout** [comment]
This is fixed in new Mongo. The output from the initial query is
```json
{ "arr": [ { "lat": 1.0 }, { "lon": 1.0 } ] }
{ "arr": [ { "lat": 20.0 }, { "lon": 30.0 } ] }
{ "arr": [ { "lat": 0.0 }, { "lon": 0.0 } ] }
```

**Comment by wemrysi** [comment]
Confirmed this is fixed in #2730.
